File: Backend/src/org/edu_sharing/service/lifecycle/PersonDeleteOptions.java
package org.edu_sharing.service.lifecycle;

import java.io.Serializable;

public class PersonDeleteOptions implements Serializable {
    public enum DeleteMode {
        none,assign,delete
    }
    public enum FolderDeleteMode {
        none,assign
    }
    public boolean cleanupMetadata;
    public HomeFolderOptions homeFolder;
    public SharedFolderOptions sharedFolders;
    public CollectionOptions collections;
    public DeleteOption ratings,comments,collectionFeedback,statistics,stream;
    public String receiver,receiverGroup;

    public static class FolderOptions {
        public FolderDeleteMode folders;
        public DeleteMode privateFiles,ccFiles;
    }
    public static class HomeFolderOptions extends FolderOptions{
        public boolean keepFolderStructure;
    }
    public static class SharedFolderOptions extends FolderOptions{
        public boolean move;
    }

    public static class CollectionOptions {
        public DeleteMode privateCollections,publicCollections;
    }

    public static class DeleteOption {
        public boolean delete;
    }
}

File: Frontend/projects/edu-sharing-api/src/lib/api/models/person-delete-options.ts
/* tslint:disable */
/* eslint-disable */
import { CollectionOptions } from './collection-options';
import { DeleteOption } from './delete-option';
import { HomeFolderOptions } from './home-folder-options';
import { SharedFolderOptions } from './shared-folder-options';
export interface PersonDeleteOptions {
    cleanupMetadata?: boolean;
    collectionFeedback?: DeleteOption;
    collections?: CollectionOptions;
    comments?: DeleteOption;
    homeFolder?: HomeFolderOptions;
    ratings?: DeleteOption;
    receiver?: string;
    receiverGroup?: string;
    sharedFolders?: SharedFolderOptions;
    statistics?: DeleteOption;
    stream?: DeleteOption;
}
